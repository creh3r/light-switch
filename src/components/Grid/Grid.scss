.grid {
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
  padding: $gutter;

  &__inner {
    border: 2px solid map-get($theme-colors, greyDarkened);
  }

  &__row {
    display: flex;
  }
}