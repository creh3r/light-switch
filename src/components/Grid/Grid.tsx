import React from 'react';
import Tile from '../Tile';

type GridProps = {
  data: number[][],
  clickHandler: (coord: number[]) => void,
};

function renderRow(row: number[], rowIndex: number, clickHandler: (coord: number[]) => void) {
  return (
    <div className="grid__row" key={rowIndex.toString()}>
      { row.map((col: number, columnIndex: number) => (
        <Tile
          key={columnIndex.toString()}
          isActive={!!col}
          clickHandler={() => clickHandler([rowIndex, columnIndex])}
        />
      ))}
    </div>
  );
}

/**
 * Render the game grid
 *
 * @param data - The game data as 2D array
 * @param clickHandler - The click handler
 */
function Grid({ data, clickHandler }: GridProps) {
  return (
    <div className="grid">
      <div className="grid__inner">
        { data.map((row: number[], rowIndex: number) => renderRow(row, rowIndex, clickHandler)) }
      </div>
    </div>
  );
}

export default Grid;
