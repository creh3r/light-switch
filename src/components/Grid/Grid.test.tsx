// eslint-disable-next-line import/no-unresolved
import { shallow } from 'enzyme';
import React from 'react';
import Grid from './Grid';

describe('<Grid /> component', () => {
  it('... should match snapshot', () => {
    const wrapper = shallow(<Grid data={[[0, 1], [1, 0]]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
