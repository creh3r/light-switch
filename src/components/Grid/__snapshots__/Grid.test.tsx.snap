// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> component ... should match snapshot and have tiles 1`] = `
<div
  className="grid"
>
  <div
    className="grid__inner"
  >
    <div
      className="grid__row"
      key="0"
    >
      <Tile
        clickHandler={[Function]}
        isActive={false}
        key="0"
      />
      <Tile
        clickHandler={[Function]}
        isActive={true}
        key="1"
      />
    </div>
    <div
      className="grid__row"
      key="1"
    >
      <Tile
        clickHandler={[Function]}
        isActive={true}
        key="0"
      />
      <Tile
        clickHandler={[Function]}
        isActive={false}
        key="1"
      />
    </div>
  </div>
</div>
`;
